### PR TITLE
Propose updates to JS guide

### PR DESCRIPTION
**Pull request by delawski**

### CODE REVIEW

File: languages/js/README.md [hunk 1]
```diff
@@ -12,11 +12,11 @@ On all new projects you should be using up to date JavaScript methodologies comb
 
 Some older projects that have not yet been upgraded may not have the capability to use the most modern techniques, but it is still important to have processes in place that allow us to grow the technology stack as a project matures. In these cases, you should still follow best practice recommendations even if the newest patterns are not yet available to you.
 
-### **Using Classes**
```
Comment (derekherman) [inline languages/js/README.md:15]: @delawski sorry I didn't see this PR until today, could you please revert the changes to the titles for now. I'm not seeing why we would change titles from say h3 to an h4 without seeing it visually on the page.

File: languages/js/README.md [hunk 2]
```diff
@@ -314,6 +314,20 @@ For various reasons on a project, you may not be able to use a modern technique
 
 Please see the [MDN XMLHttpRequest documentation](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) for an example of a basic Ajax call.
 
+### Using `@wordpress/api-fetch` for WordPress REST API
+
+When dealing with WordPress REST API it may be convenient to use the [`apiFetch`](https://developer.wordpress.org/block-editor/packages/packages-api-fetch/) utility. It is a think wrapper around the native `window.fetch` function.
```
Comment (delawski) [inline languages/js/README.md:319]:
```suggestion
When dealing with WordPress REST API it may be convenient to use the [`apiFetch`](https://developer.wordpress.org/block-editor/packages/packages-api-fetch/) utility. It is a thin wrapper around the native `window.fetch` function.
```

File: languages/js/README.md (hunk 1 above)
Comment (delawski) [reply languages/js/README.md:15]: @derekherman Sure thing. I'll move those formatting changes over to a separate PR and provide some visuals so that it's clear what the diff is.

Comment (mehigh) [reply languages/js/README.md:15]: Thanks! Ping the PR when that is completed.
